src/ol/events/condition.js

```diff
@@ -13,6 +13,29 @@ import {assert} from '../asserts.js';
  * @typedef {function(this: ?, import("../MapBrowserEvent.js").default): boolean} Condition
  */
 
+/**
+ * Creates a condition function that is only fulfilled when a chain of conditions pass.
+ * @param {...Condition} var_args Conditions to check.
+ * @return {Condition} Condition function that checks a chain of conditions.
+ */
+export function chain(var_args) {
```

Comment:
If we want to make this part of the API in the future, I'd be tempted to name it `all` or `and`.